src/analysis/modal-utils.ts

```diff
@@ -33,61 +39,136 @@ export interface IComputeConfig {
+  bootDiskSize?: number | null;
+  diskSize?: number;
+  componentGatewayEnabled?: boolean;
+  masterMachineType?: string;
+  masterDiskSize?: number;
+  numberOfWorkers?: number;
+  numberOfPreemptibleWorkers?: number | null;
+  workerMachineType?: string | null | undefined;
+  workerDiskSize?: number | null | undefined;
+  timeoutInMinutes?: number | null;
+};
+
+// Auxiliary functions -- begin
+export const isGce = (runtimeType) => runtimeType === runtimeTypes.gceVm;
+
+export const isDataproc = (runtimeType) =>
+  runtimeType === runtimeTypes.dataprocSingleNode || runtimeType === runtimeTypes.dataprocCluster;
+
+export const isDataprocCluster = (runtimeType) => runtimeType === runtimeTypes.dataprocCluster;
+
+// TODO: use type
+export const runtimeTypes = {
+  gceVm: 'Standard VM',
+  dataprocSingleNode: 'Spark single node',
+  dataprocCluster: 'Spark cluster',
+};
+
+export const shouldUsePersistentDisk = (runtimeType, runtimeDetails, upgradeDiskSelected) =>
+  isGce(runtimeType) && (!runtimeDetails?.runtimeConfig?.diskSize || upgradeDiskSelected);
+
+// TODO: tnis is the best place to start
 export const buildExistingEnvironmentConfig = (
   computeConfig: IComputeConfig,
-  currentRuntimeDetails: any,
-  currentPersistentDiskDetails: PersistentDisk,
-  isDataproc: boolean
-) => {
+  currentRuntimeDetails?: GetRuntimeItem,
+  currentPersistentDiskDetails?: PersistentDisk,
+  isDataproc = false
+): ExistingModalRuntimeConfig => {
   const runtimeConfig = currentRuntimeDetails?.runtimeConfig;
   const cloudService = runtimeConfig?.cloudService;
-  const numberOfWorkers = runtimeConfig?.numberOfWorkers || 0;
-  const hasGpu = computeConfig.hasGpu;
-  const gpuConfig = runtimeConfig?.gpuConfig;
-  const toolLabel = getToolLabelFromCloudEnv(currentRuntimeDetails);
+  const numberOfWorkers = runtimeConfig && 'numberOfWorkers' in runtimeConfig ? runtimeConfig.numberOfWorkers : 0;
+  const gpuConfig = runtimeConfig && 'gpuConfig' in runtimeConfig ? runtimeConfig.gpuConfig : undefined;
+  const toolLabel = currentRuntimeDetails ? getToolLabelFromCloudEnv(currentRuntimeDetails) : undefined;
   return {
-    hasGpu,
+    hasGpu: computeConfig.hasGpu,
     autopauseThreshold: currentRuntimeDetails?.autopauseThreshold,
+    computeConfig,
+    currentRuntimeDetails,
     runtime: currentRuntimeDetails
       ? {
           cloudService,
           toolDockerImage: getImageUrlFromRuntime(currentRuntimeDetails),
           tool: toolLabel,
+          region: computeConfig.computeRegion,
+          persistentDiskAttached: !!(runtimeConfig && 'persistentDiskId' in runtimeConfig),
+          gpuConfig,
           ...(currentRuntimeDetails?.jupyterUserScriptUri
             ? {
                 jupyterUserScriptUri: currentRuntimeDetails?.jupyterUserScriptUri,
               }
             : {}),
-          ...(currentRuntimeDetails?.timeoutInMinutes
-            ? {
-                timeoutInMinutes: currentRuntimeDetails?.timeoutInMinutes,
-              }
```

Comment:
no, it is never used off the existing runtime config object

Reply:
Instead of removing this, it'd be better to fix the bug where this isn't used, and loading a runtime with a custom timeout does not display the correct value - GcpComputeModal line 236 has `const [timeoutInMinutes, setTimeoutInMinutes] = useState(null);` - this could change to `useState(currentRuntime?.timeoutInMinutes || null`

Reply:
or at least leaving a comment on that line with the needed change

Reply:
this isn't actually a field on existing runtimes! I think the types demonstrate well that this is a field for `desired` runtime that we submit in the request, but it is not something leo returns back on `existing`.

Reply:
I'll add a comment on the type